accounts/abi/event.go

```diff
@@ -69,16 +69,21 @@ func (e Event) tupleUnpack(v interface{}, output []byte) error {
 	for i := 0; i < len(e.Inputs); i++ {
 		input := e.Inputs[i]
 		if input.Indexed {
+			// indexed inputs are not available in log output
+			j--
 			// can't read, continue
 			continue
-		} else if input.Type.T == ArrayTy {
-			// need to move this up because they read sequentially
-			j += input.Type.Size
 		}
 		marshalledValue, err := toGoType((i+j)*32, input.Type, output)
 		if err != nil {
 			return err
 		}
+		if input.Type.T == ArrayTy {
```

Comment:
In current codebase array fields are counted before unpacking output.
So, if we have array with 2 fields: toGoType function will try to unpack
values from 64 till 64 + 2*32 which is not correct in my opinion.

I shifted counting to be done after processing array, so in case array is the first
field, and it has 2 elements: we will read it from 0 till 2*32.

Additionally array doesnt require length prefix, so field that follows array
should be read from 64 byte in described case.

Reply:
I was using this event for test
`event Echo(uint8[2] message, address from);`

error message
`
2017/11/30 12:37:46 abi: cannot marshal in to go array: offset 96 would go over slice boundary (len=128)
`

Reply:
Why would that be incorrect? Sounds right at the beginning.

Reply:
how is that correct? example that i provided actually fails on current master

let's say we will try to unpack an event with ([7,8], 9) all simple integers,
it will be encoded as 
```
0000000000000000000000000000000000000000000000000000000000000007
0000000000000000000000000000000000000000000000000000000000000008
0000000000000000000000000000000000000000000000000000000000000009
```
without this change we will try to read array from 64 till 128, and integer from 96 till 128, and actually will fail with this error `2017/12/07 16:04:29 abi: cannot marshal in to go array: offset 96 would go over slice boundary (len=128)`

with this change first read will be done from 0 till 64 (array), and for integer from 64 till 96

Reply:
Ah. Okay. Yea I'm sorry for the wait, I've reviewed the code, it looks fairly good, but I would just add a comment about the `j += input.Type.Size - 1`...specifically why you're subtracting the 1. Just leave a comment on that and I'll approve this...for whatever that's worth.

Reply:
@VoR0220 added a comment